src/main/java/com/ably/kafka/connect/client/DefaultAblyClient.java

```diff
@@ -53,10 +53,10 @@ public void connect(SuspensionCallback suspensionCallback) throws ConnectExcepti
                     connectionFailed.set(true);
                 } else if (connectionStateChange.current == ConnectionState.connected) {
                     logger.info("Ably connection successfully established");
-                    suspensionCallback.on(false);
+                    suspensionCallback.onSuspendedStateChange(false);
                 } else if (connectionStateChange.current == ConnectionState.suspended) {
                     logger.info("Ably connection is suspended");
-                    suspensionCallback.on(true);
+                    suspensionCallback.onSuspendedStateChange(true);
```

Comment:
Normal messages are discarded upon entering the suspended state, the point of the work we're doing here is for the kafka connector to queue messages while suspended since the ably-java client doesn't do that itsself

Reply:
If normal messages are discarded from the Ably client lib's queue upon entering the suspended state, would that mean any messages sent during the Disconnected state would be lost with this implementation? So like we'd have:

* messages are sent correctly during the Connected state
* messages get queued during the disconnected state in the Ably client lib's queue
* those are discarded as the client enters the Suspended state, and then further messages sent during this Suspended state are added to this new queue for the Connector, to eventually be sent once we enter the Connected state again

If that were the case, we'd lose the messages from the Disconnected state I believe, if they are discarded upon entering the Suspended state.

Having said that, some cursory testing of this in the JS library seems to indicate the messages are not in fact discarded from the Ably Client library's queue when going from Disconnected -> Suspended, so if that were the cast then all is well!

Reply:
Ah no you're totally right, I forgot that pending messages are queued separately from pending protocol messages, so yeah we should be fine here. thank you for checking :)